obspy/clients/fdsn/routers/__init__.py

```diff
@@ -0,0 +1,210 @@
+includes geographic parameters. For more details, see the help for
+obspy.clients.fdsn
+
+    >>> from obspy import UTCDateTime
+    >>> t = UTCDateTime("2010-02-27T06:45:00.000")
+    >>> st = client.get_waveforms("IU", "ANMO", "00", "LHZ", t, t + 60 * 60)
+    >>> st.plot()  # doctest: +SKIP
+
+    .. plot::
+
+        from obspy import UTCDateTime
+        from obspy.clients.fdsn import FederatedClient
+        client = Client()
+        t = UTCDateTime("2010-02-27T06:45:00.000")
+        st = client.get_waveforms("IU", "ANMO", "00", "LHZ", t, t + 60 * 60)
+        st.plot()
+
+caveats
+----
+* duplicated metadata remains duplicated.
+
+suggestions for future
+----------------------
+* bulk requests for a certain level of metadata, eg. network, station
+  will return an egregious amount of data that is very likely duplicated
+  Perhaps the fedcatalog web service could recognize it and return pared
+  down data or the fdsn.client.Client  class could recognize and reduce
+  the duplications. example:
+    >>> INV2 = fclient.get_stations(network="I?", station="AN*",
+    ...                           level="network", includeoverlaps="true")
+    ...                           #doctest: +SKIP
+
+
+Getting comfortable with the building blocks of the routing modules:
+The simplest unit of a request is the FDSNBulkRequestItem.
+
+FDSNBulkRequestItem
+------------------------------------------------------
+The FDSNBulkRequestItem represents one bulk request line, containing the
+following information:
+
+  network station location channel starttime endtime
+
+in a format that might look like A:
+
+A)  IU ANMO 00 BHZ 2015-05-24T12:00:00 2015-05-24T12:05:00
+
+It could include wildcards in any of the fields, as shown in B...
+
+B)  IU ANMO * * 2015-05-24T12:00:00 *
+
+>>> from obspy.clients.fdsn.routers.fedcatalog_parser import \
+        FDSNBulkRequestItem
+
+Using obspy.clients.fdsn.routers.FDSNBulkRequestItem, each could be created as:
+>>> itemA = FDSNBulkRequestItem('IU ANMO 00 BHZ 2015-05-24T12:00:00 '
+...                             '2015-05-24T12:05:00')
+
+And, item B, with wildcards could be created similarly.
+>>> itemB = FDSNBulkRequestItem('IU ANMO * * 2015-05-24T12:00:00 *')
+
+  or, with parameters...
+>>> itemB = FDSNBulkRequestItem(network='IU', station='ANMO',
+...                             starttime='2015-05-24T12:00:00')
+
+  or, using obspy.core.UTCDateTime ...
+>>> t1 = UTCDateTime(2015, 5, 24, 12, 0, 0)
+>>> itemB = FDSNBulkRequestItem(network='IU', station='ANMO', starttime=t1)
+
+Basic comparisons can be made between these items.
+A < B : alphabetically compares net, stations, locations, then chans,
+        and then starttime by date.
+A contains B : This takes wildcards and time ranges into account to denote
+               whether a request for A would include B's data too.
+A == B : all fields are the same (wildcards only match wildcards, for example)
```

Comment:
This whole section renders pretty awkward in sphinx, can you please try to fix it? http://docs.obspy.org/pull-requests/1779/packages/autogen/obspy.clients.fdsn.routers.html#module-obspy.clients.fdsn.routers

![screenshot from 2017-10-04 17-43-14](https://user-images.githubusercontent.com/1842780/31185145-891e0a22-a92b-11e7-9fe5-df621c8d1dad.png)